src/J2N/Text/CharArrayExtensions.cs

```diff
@@ -265,8 +265,10 @@ public static int CompareToOrdinal(this char[]? str, ICharSequence? value)
         /// </returns>
         public static int CompareToOrdinal(this char[]? str, ReadOnlySpan<char> value)
         {
+#pragma warning disable CA2265 // Do not compare Span<T> to null or default
             if (str is null) return (value == default) ? 0 : -1;
             if (value == default) return 1;
```

Comment:
Are you sure these `default` checks should be done? It might not give the results that are expected, which is perhaps why the CA2265 warning exists.

For example, `string.Empty.AsSpan()` does not equal `ReadOnlySpan<char>.Empty`, so if `str is null` then it will return `0` for `string.Empty.AsSpan()` and `-1` for `ReadOnlySpan<char>.Empty` as the value argument. The only thing that would be caught by `== default` is if it's _exactly_ `ReadOnlySpan<char>.Empty` and not an empty string as a span. That makes the ternary on line 269 suspicious, and `return 1` case likely an unnecessary (or at best, very rare) optimization, given that most people are probably going to be passing in things that are converted to span to this method rather than `ReadOnlySpan<char>.Empty`. This also applies to StringBuilderExtensions and StringExtensions in this PR.

Reply:
Also I was referring to strings above, but that also applies to char arrays. `Array.Empty<char>().AsSpan() != ReadOnlySpan<char>.Empty` (or `default`) holds as well.

Reply:
You are right to question this, but what we are doing is filtering out the cases where the value was intentionally left as default before running the actual comparison business logic. The cases for empty string are handled in the rest of the code (checking length, etc.)

So, what we are doing is making `value == default` equivalent to `str == default` even though they don't compare equally. We know if those two are specified that the two cases are equal so we can return 0.

With this line:

```c#
if (value == default) return 1;
```
we know the string is not `null`, but that is all we know about it. If it is empty it would technically match. However, if the `ReadOnlySpan<char>` is exactly default, we should be returning a 1. If it were instead meant to be an empty string, then it should bypass this and be compared with the rest of the logic.

This is where things are a bit unclear. In Java, this would have been allowed to throw an exception if one of the values were `null` and there is no such thing as a user defined type that is not a reference type. However, in .NET `Compare()` methods will never throw exceptions when a value is `null`. They always have these checks to return the right value if one, the other or both parameters are `null`.

https://github.com/dotnet/runtime/blob/c0b52f635224880fb6437c4b328a3ad39c927b19/src/libraries/System.Private.CoreLib/src/System/Collections/Comparer.cs#L54-L56

So, this is an oddball because it can technically never be `null` to compare against a `null` string. But it can have a `default`, so we are using that instead. Empty is a completely different case.

But I could be wrong. Maybe there is a compelling reason to leave out the `null` check on the right side and proceed to always compare them as empty. To me it seems like an uninitialized `Span<char>` or `ReadOnlySpan<char>` is passed in, it should be treated as if it were `null`.

Basically, the choice here is whether to leave it as-is or change it to:

```c#
if (str is null) return -1;
```

and then treat `default` the same as empty on the right side.

Reply:
Alright, I checked with ChatGPT and it gave a compelling argument why we should not be using `default` on the right side from the perspective of how Microsoft designs comparison APIs.

https://chatgpt.com/share/673e30d0-6a54-8005-be1f-0f6b207c5e1d

That and the fact there is a warning seem to both agree that changing this to consider `default` to be the same as `Empty` would be the right design choice.

Reply:
I opened #123 so we can fix this in J2N 3.0.